### PR TITLE
py-gfal2-python: new package to fix gfal2-util

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds `py-gfal2-python` for use in `gfal2-util`. Apologies for the duplicative name for `py-gfal2-python` but that's out of my control. 

Test builds:
```
-- linux-ubuntu24.10-skylake / gcc@14.2.0 -----------------------
nbf3aw6 gfal2-util@1.9.0 build_system=python_pip  2o3hzoz py-gfal2-python@1.13.0 build_system=python_pip
==> 2 installed packages
```

Test run:
```
$ gfal-ls root://dtn-eic.jlab.org//work/eic2/
transfer
ATHENA
eicbreakdown.gz
DET1
eic2_user_usage
ruciotest
EPIC
CORE
mc
user_usage
users
```

### CODE REVIEW

**Comment by wdconinc** [comment]
@tmadlener Are you all using gfal2 utils for rucio writes? Would you want to give this a review?

File: var/spack/repos/builtin/packages/py-gfal2-python/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    version("1.13.0", sha256="5be42cc894fa20af3d6f6dbb30dfd4d29ab49bd5f15b3e3e754aa25c5ed17997")
+
+    depends_on("cxx", type="build")
+    depends_on("python", type="build")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("cmake", type="build")
+
+    depends_on("boost")
```
Comment (tmadlener) [inline var/spack/repos/builtin/packages/py-gfal2-python/package.py]: Any chance on finding out the actual variants / components of boost that are necessary?

**Review by tmadlener — COMMENTED** [review]
I am not entirely sure if we are using rucio at all at the moment. It might be hidden inside iLCDirac @andresailer? Given that I basically can't really contribute much to the review here other then the boost variants below.

File: var/spack/repos/builtin/packages/py-gfal2-python/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-gfal2-python/package.py]: Good catch.
```suggestion
    depends_on("boost +python")
```

**Review by tldahlgren — APPROVED** [review]
Confirmed the homepage and version sha256 of the new package.

**Pull request merged**